### PR TITLE
WebSocket support #173

### PR DESCRIPTION
**Pull request by StormHub**

### CODE REVIEW

File: examples/WebSockets.Server/WebSocketServerBenchmarkPage.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+{
+    using System.Text;
+    using DotNetty.Buffers;
+
+    static class WebSocketServerBenchmarkPage
+    {
+        const string Newline = "\r\n";
+
+        public static IByteBuffer GetContent(string webSocketLocation) =>
+            Unpooled.CopiedBuffer(
```
[the remark this answers is not in the document]
Comment (StormHub) [reply examples/WebSockets.Server/WebSocketServerBenchmarkPage.cs]: Following Netty one
https://github.com/netty/netty/blob/4.1/example/src/main/java/io/netty/example/http/websocketx/benchmarkserver/WebSocketServerBenchmarkPage.java#L30

[the remark this answers is not in the document]
Comment (nayato) [reply examples/WebSockets.Server/WebSocketServerBenchmarkPage.cs]: Not exactly. See my next comment on this.

File: src/DotNetty.Codecs.Http/WebSockets/TextWebSocketFrame.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        static IByteBuffer FromText(string text) => string.IsNullOrEmpty(text) 
+            ? Unpooled.Empty : Unpooled.CopiedBuffer(Encoding.UTF8.GetBytes(text));
+
+        public TextWebSocketFrame(bool finalFragment, int rsv, IByteBuffer binaryData)
+            : base(finalFragment, rsv, binaryData)
+        {
+        }
+
+        public string Text => this.Content.ToString(Encoding.UTF8);
```
[the remark this answers is not in the document]
Comment (StormHub) [reply src/DotNetty.Codecs.Http/WebSockets/TextWebSocketFrame.cs]: Changed all to use https://github.com/Azure/DotNetty/blob/dev/src/DotNetty.Buffers/Unpooled.cs#L341

File: examples/WebSockets.Server/WebSocketServerBenchmarkPage.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (StormHub) [reply examples/WebSockets.Server/WebSocketServerBenchmarkPage.cs]: Yep, i saw it.

**Pull request merged**